src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs

```diff
@@ -55,6 +65,14 @@ protected override void ExecuteCore()
                     }
                 }
 
+                // For any RuntimeFrameworks that matches with FrameworkReferences, we can apply that RuntimeFrameworks profile to the corresponding RuntimePack
+                // Matches the RuntimeFramework with the RuntimePack by comparing the FrameworkName metadata. 'profile' will be an empty string if no matching RuntimeFramework is found
+                string profile = matchingRuntimeFrameworks?
```

Comment:
Rather than join multiple profiles with semicolons, I think this should just be a `List<string>` (or `HashSet<string>`, since later we're doing a contains check on it.